addons/mail/static/src/discuss/core/common/discuss_core_common_service.js

```diff
@@ -120,7 +120,12 @@ export class DiscussCoreCommon {
                     thread: { id: channel_id, model: "discuss.channel" },
                 });
                 if (member?.persona.eq(this.store.self)) {
-                    this.threadService.updateSeen(member.thread, last_message_id);
+                    // Wait for `discuss.channel/new_message` to be processed to
+                    // ensure the message is in the thread.
+                    await new Promise(setTimeout);
```

Comment:
As we discussed, for master, I think we could bundle the `updateSeen` notification inside the `new_message` notification to avoid network issues like this one.